documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,258 @@ Control characters are not allowed in a specification.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3008" />R3008 CollectionObjectPropertiesNaming
+**Output Message**: Collection object '{0}' returned by list operation '{1}' with 'x-ms-pageable' extension, has no property named 'value'. 
+
+**Description**: Per ARM guidelines, a model returned by an `x-ms-pageable` operation must have a property named `value`. This property indicates what type of array the object is.
+
+**Why the rule is important**: To maintain consistency on how `x-ms-pageable` operations and corresponding response objects are modeled and to enable execution of other validation rules based on this consistent structure. More documentation about the extension can be found [here](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-pageable).
+
+**How to fix the violation**: Ensure that the response object has a property named `value` of `array` type.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2027" />R2027 DefaultMustBeInEnum
+**Output Message**: The default value is not one of the values enumerated as valid for this element.
+
+**Description**: The value assigned as a default for an enum property must be present in the enums' list.
+
+**Why the rule is important**: Generated SDKs in types languages may fail to compile if we try to enforce a default value that is not a part of the enums defined in the list and in other languages may fail in serialization/deserialization phases.
+
+**How to fix the violation**: Ensure that the default desired actually exists in the enums' list.
+
+**Bad Example**:
+```
+"status":{
+  "type": "string",
+  "enum": [
+    "Succeeded",
+    "Updating",
+    "Deleting",
+    "Failed"
+  ],
+  "default": "Terminated"
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2047" />R2047 NamePropertyDefinitionInParameter
+**Output Message**: Parameter Must have the "name" property defined with non-empty string as its value.
+
+**Description**: A parameter must have a `name` property for the SDK to be properly generated. 
+
+**Why the rule is important**: AutoRest fails to generate code if the `name` property is not provided for a parameter.
+
+**How to fix the violation**: Add a non-empty `name` property to the parameter.
+
+**Bad Example**:
```

Comment:
maybe we should directly have a good example in this case, instead of a bad one.